### PR TITLE
Add injected sidecar test

### PR DESCRIPTION
**Pull request by kevinwangcn**
# Intent

Test resize/move action execution on a deployment with injected sidecar container in a Istio environment

# Background
In a typical Istio environment,  the pod will be injected a sidecar container with the name `istio-proxy`, kubeturbo has to handle this situation well. 

# Implementation
1)Deploy Istio on the Kind cluster, when testing this case, patch the namespace with the label `"istio-injection": "enabled"`, trigger the Istio webhook inject the sidecar container into the pod
2)Validate if the pod has the sidecar container, if not, fail the case
3)If yes, compose the resize/move action, execute the action against the deployment
4)Validate if the pod is moved or resized as expected

# Test Done

1. Run the command to start the test 
```
./build/integration.test -k8s-kubeconfig=$HOME/.kube/config -k8s-context=kind-kind -istio-enabled=true -ginkgo.focus=Istio -ginkgo.focus=teardown
```
2. Check the result , it includes 3 test specs, 1 for move action, 1 for resize action, and the last one for tear down.
```
I0615 16:50:12.940648   21634 integration_test.go:52] Starting integration run on Ginkgo node 1
Running Suite: Kubeturbo integration suite
==========================================
Random Seed: 1655326212
Will run 3 of 20 specs

SE0615 16:50:19.630799   21634 util.go:96] cannot found node name from properites: []
W0615 16:50:19.634718   21634 rescheduler.go:153] Move action: pod[kubeturbo-test-action-executor-hbgcf/test-nrq97-987b95586-nm4mn]'s new host(kind-worker) in bad condition: MemoryPressure
W0615 16:50:19.634732   21634 rescheduler.go:153] Move action: pod[kubeturbo-test-action-executor-hbgcf/test-nrq97-987b95586-nm4mn]'s new host(kind-worker) in bad condition: DiskPressure
W0615 16:50:19.634735   21634 rescheduler.go:153] Move action: pod[kubeturbo-test-action-executor-hbgcf/test-nrq97-987b95586-nm4mn]'s new host(kind-worker) in bad condition: PIDPressure

------------------------------
• [SLOW TEST:16.875 seconds]
Action Executor 
/opt/git/kubeturbo/test/integration/action_execution.go:56
  executing action move pod with volume attached, will also be tested in Istio environment
  /opt/git/kubeturbo/test/integration/action_execution.go:122
    should result in new pod on target node
    /opt/git/kubeturbo/test/integration/action_execution.go:123
------------------------------
SSSSS•S
------------------------------
• [SLOW TEST:53.833 seconds]
Action Executor 
/opt/git/kubeturbo/test/integration/action_execution.go:56
  test teardown
  /opt/git/kubeturbo/test/integration/action_execution.go:381
    Deleting framework namespace: 
    /opt/git/kubeturbo/test/integration/action_execution.go:382
------------------------------
SSSSSSSSSS
Ran 3 of 20 Specs in 75.242 seconds
SUCCESS! -- 3 Passed | 0 Failed | 0 Pending | 17 Skipped
PASS
```

### CODE REVIEW

**Comment by irfanurrehman** [comment]
@kevinwangcn as talked on slack, please check if you can simply use -ginkgo.focus "regular expr for your specs "
[11:49](https://turbonomic.slack.com/archives/D035BEFLYLF/p1655257774808389)
- ./build/integration.test -k8s-kubeconfig=$HOME/.kube/config -k8s-context=kind-kind
  - ./build/integration.test -k8s-kubeconfig=$HOME/.kube/config -k8s-context=kind-kind --istio-enabled -ginkgo.focus "regular expr for your specs "
  - ./build/integration.test -k8s-kubeconfig=$HOME/.kube/config -k8s-context=ocp47 --openshift-tests=true

**Comment by kevinwangcn** [comment]
> @kevinwangcn as talked on slack, please check if you can simply use
> 
> `-ginkgo.focus "regular expr selecting your specs "`
> 
> ```
> - ./build/integration.test -k8s-kubeconfig=$HOME/.kube/config -k8s-context=kind-kind
>   - ./build/integration.test -k8s-kubeconfig=$HOME/.kube/config -k8s-context=kind-kind --istio-enabled -ginkgo.focus "regular expr selecting your specs "
>   - ./build/integration.test -k8s-kubeconfig=$HOME/.kube/config -k8s-context=ocp47 --openshift-tests=true
> ```

Implement this in the new code change, please have a look @irfanurrehman . thanks.

File: test/integration/framework/test_context.go [hunk 1]
```diff
@@ -32,6 +33,8 @@ func registerFlags(t *TestContextType) {
 		fmt.Sprintf("The maximum duration of a single call.  If unset, will default to %v", DefaultSingleCallTimeout))
 	flag.BoolVar(&t.IsOpenShiftTest, "openshift-tests", false,
 		"If set, the test will only run the Openshift case. By default, it's set to false, only the non-Openshift cases get run")
+	flag.BoolVar(&t.IsIstioEnabled, "istio-enabled", false,
```
Comment (enlinxu) [inline test/integration/framework/test_context.go:36]: Why are we disabling it by default?

Comment (kevinwangcn) [reply test/integration/framework/test_context.go:36]: we only open it when we want to test istio case,  @enlinxu , please have a look at the change on `.travis.yml`. We added a line there, by default, we only test the regular cases, the namespace isn't injected. When we want to test sidecar cases, we set it to ture and specify the parameter `-ginko.focus` to filter out the cases.  In such way,  we're trying to reuse the code without adding more resize&move cases for sidecar.

**Review by pankajbatra1 — APPROVED** [review]

**Pull request merged**